### DISCUSSION TITLE
How to wrap real html child after remove buildAsyncBuilder

### DISCUSSION THREAD

**Discussion by ykrank** [daohoangson/flutter_widget_from_html#595] [Q&A] [answered]
## Use case

Before, i use buildAsyncBuilder to get real html height, like:
```dart
buildAsyncBuilder: (context, snapshot) => isDebug && snapshot.hasError
                  ? Container(
                      child: Text(snapshot.error.toString()),
                    )
                  : snapshot.hasData
                      ? MeasureSize(
                          onChange: (Size? size) {
                            // print('size ${size?.height}');
                            if (size != null) _post.extraCache.measureHeight = size.height;
                          },
                          child: snapshot.data)
                      : Container(
                          height: _post.extraCache.measureHeight,
                          child: Loading(),
                        ),
```
but now i can't find a entrypoint to do this.
How can i do this like before?

**Comment by daohoangson** [comment]
There is no direct replacement for this. The `onErrorBuilder` and `onLoadingBuilder` doesn't have access to the rendered widget anymore. If you need the height, it should be possible to measure the entire `HtmlWidget`.

**Comment by ykrank** [comment]
> There is no direct replacement for this. The `onErrorBuilder` and `onLoadingBuilder` doesn't have access to the rendered widget anymore. If you need the height, it should be possible to measure the entire `HtmlWidget`.

but entire HtmlWidget have padding or margin, this make it unstable

**Comment by daohoangson** [comment, marked as answer]
The recent versions no longer have built-in paddings, I think it should be usable.
